### PR TITLE
Update: [AEA-4048] - Validate terminal status indicator value

### PR DESCRIPTION
**Pull request by kris-szlapa**
## Summary

🎫 [AEA-4048](https://nhsd-jira.digital.nhs.uk/browse/AEA-4048) Validate terminal status indicator value

- Routine Change

### Details

Validate terminal status indicator value

### CODE REVIEW

File: packages/updatePrescriptionStatus/tests/validation/testRequestContentValidation.test.ts [hunk 1]
```diff
@@ -211,34 +211,80 @@ describe("Unit tests for validation of NHS number", () => {
...
-      expected: undefined
-    },
-    {
-      taskStatus: "in-progress",
-      businessStatus: "Ready to collect",
-      expected: undefined
-    }
-  ])(
-    "When status is '$status' and business status is '$businessStatus', should return expected issue.",
+  const generateTestCase = (taskStatus: string, businessStatus: string, expected: string | undefined) => ({
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/tests/validation/testRequestContentValidation.test.ts:214]: I think this test needs splitting, to make it clearer what the intention is.

One for each taskStatus, I'd say.

File: packages/updatePrescriptionStatus/src/validation/content.ts [hunk 2]
```diff
@@ -113,15 +122,31 @@ export function businessStatus(task: Task): string | undefined {
...
+
+  if (!businessStatus) {
+    return
+  }
+
+  const coding: Coding = businessStatus.coding![0]
+  const code = coding.code
+
+  if (!code) {
+    return
```
Comment (originalphil) [inline packages/updatePrescriptionStatus/src/validation/content.ts:135]: I think we agreed in elab that we should ensure that the businessStatus is one that we expect.

The businessStatus, coding and code are all marked as required in the spec. So, we want to return a message if they don't exist. That is something I should have covered earlier, sorry.

File: packages/updatePrescriptionStatus/tests/validation/testRequestContentValidation.test.ts (hunk 1 above)
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/tests/validation/testRequestContentValidation.test.ts:212]: The test was split based on taskStatus.

File: packages/updatePrescriptionStatus/src/validation/content.ts (hunk 2 above)
Comment (kris-szlapa) [reply packages/updatePrescriptionStatus/src/validation/content.ts]: If the businessStatus, coding, or code field is missing, the following message is returned.

```
{
    "resourceType": "Bundle",
    "type": "transaction-response",
    "entry": [
        {
            "response": {
                "status": "400 Bad Request",
                "outcome": {
                    "resourceType": "OperationOutcome",
                    "meta": {
                        "lastUpdated": "2024-05-30T07:37:41.403Z"
                    },
                    "issue": [
                        {
                            "code": "processing",
                            "severity": "error",
                            "diagnostics": "Missing required field(s) - Status."
                        }
                    ]
                }
            },
            "fullUrl": "urn:uuid:4d70678c-81e4-4ff4-8c67-17596fd0aa46"
        }
    ]
}
```

**Review by originalphil — APPROVED** [review]

**Pull request merged**